src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Utils/InternalMetadataProviderIdentityExtensions.cs

```diff
@@ -34,13 +34,13 @@ public static string GetPackageName(this IInternalMetadataProvider internalMetad
             }
 
             // First check if the user provided a package name.
-            if (internalMetadataProvider.TryGetMetadata(MetadataKey.PackageName, out object packageName))
+            if (internalMetadataProvider.TryGetMetadata(MetadataKey.PackageName, out object packageName) && !string.IsNullOrWhiteSpace(packageName as string))
```

Comment:
I know you didn't necessarily write the original code here and are just amending it, but looking at it, I think there's a chance for some cleanup.

Since so much of this is copy/paste, maybe you can create a method that encapsulates the Try and whitespace check.

This is a total guess but maybe an extension method on the internalmetadataprovider that's something that has a signature like:

`public bool TryGetMetadata<T>(this InternalMetadataProvider provider, string keyName, out T returnVal)`

Reply:
@ksigmund the change has been made, if it looks good, please resolve the comment